certora/specs/GovernorStates.spec

```diff
@@ -68,18 +57,26 @@ rule stateTransitionWait(uint256 pId, env e1, env e2) {
     require clockSanity(e2);
     require clock(e2) > clock(e1);
 
+    // Force the state to be consistent with e1 (before). We want the storage related to `pId` to match what is
+    // possible before the time passes. We don't want the state transition include elements that cannot have happened
+    // before e1. This ensure that the e1 → e2 state transition is purelly a consequence of time passing.
+    requireInvariant votesImplySnapshotPassed(e1, pId);
+
     uint8 stateBefore = state(e1, pId);
     uint8 stateAfter  = state(e2, pId);
 
     assert (stateBefore != stateAfter) => (
-        stateBefore == PENDING()   => stateAfter == ACTIVE() &&
-        stateBefore == ACTIVE()    => (stateAfter == SUCCEEDED() || stateAfter == DEFEATED()) &&
-        stateBefore == UNSET()     => false &&
-        stateBefore == SUCCEEDED() => false &&
-        stateBefore == QUEUED()    => false &&
-        stateBefore == CANCELED()  => false &&
-        stateBefore == DEFEATED()  => false &&
-        stateBefore == EXECUTED()  => false
+        (stateBefore == PENDING() && stateAfter == ACTIVE()   ) ||
+        (stateBefore == PENDING() && stateAfter == DEFEATED() ) ||
+        (stateBefore == ACTIVE()  && stateAfter == SUCCEEDED()) ||
+        (stateBefore == ACTIVE()  && stateAfter == DEFEATED() ) ||
+        // Strange consequence of the timelock binding:
+        // When transitioning from ACTIVE to SUCCEEDED (because of the clock moving forward) the proposal state in
+        // the timelock is suddenly considered. Prior state set in the timelock can cause the proposal to already be
+        // queued, executed or canceled.
+        (stateBefore == ACTIVE()  && stateAfter == CANCELED()) ||
+        (stateBefore == ACTIVE()  && stateAfter == EXECUTED()) ||
+        (stateBefore == ACTIVE()  && stateAfter == QUEUED())
```

Comment:
Not sure what we should do here. This rule is currently not looking good, as we would never expect a proposal to go (for example) from pending to defeated automatically, I think...

If the rule was working before (using `=>`) we should probably keep it like that.

Reply:
I needed to add an additional "helper" invariant, but I managed to fix that part :)